app/services/discord/sync_profile_service.rb

```diff
@@ -1,42 +1,147 @@
 module Discord
   class SyncProfileService
-    def initialize(user)
+    attr_reader :error_msg, :warning_msg
+
+    def initialize(user, additional_discord_role_ids: nil)
       @user = user
+      @additional_discord_role_ids =
+        if additional_discord_role_ids.is_a?(Array)
+          additional_discord_role_ids
+        else
+          [additional_discord_role_ids].compact
+        end.map(&:to_s)
+
+      @error_msg = ""
+      @warning_msg = ""
     end
 
     def execute
-      return unless @user.discord_user_id.present? && configuration.configured?
-
-      role_ids = [
-        @user.cohorts.pluck(:discord_role_ids),
-        configuration.default_role_ids
-      ].flatten.compact.uniq
-
-      return if role_ids.empty?
-
-      Discordrb::API::Server.update_member(
-        "Bot #{configuration.bot_token}",
-        configuration.server_id,
-        @user.discord_user_id,
-        roles: role_ids,
-        nick: nick_name
-      )
+      return false unless sync_ready?
```

Comment:
This is an unusual case. It looks like this is handled in `Schools::UsersController#L88`, inside the `else` block with a very generic error message that expects `sync_service.error_msg` to return something... but `@error_msg` isn't set here, so the generic error message is unhelpful. I think this counts as an error.

What's probably appropriate is to `raise SyncError` here (as detailed in the comment in `SyncProfileService`) with a message similar to `Failed to sync! Either the user doesn't have Discord account connected, or the Discord integration is currently missing.`